asserts/crypto.go

```diff
@@ -104,54 +108,43 @@ func signContent(content []byte, privateKey PrivateKey) ([]byte, error) {
 		return nil, err
 	}
 
-	return encodeFormatAndData("openpgp", buf.Bytes()), nil
-}
-
-func splitFormatAndBase64Decode(formatAndBase64 []byte) (string, []byte, error) {
-	parts := bytes.SplitN(formatAndBase64, []byte(" "), 2)
-	if len(parts) != 2 {
-		return "", nil, fmt.Errorf("expected format and base64 data separated by space")
-	}
-	buf := make([]byte, base64.StdEncoding.DecodedLen(len(parts[1])))
-	n, err := base64.StdEncoding.Decode(buf, parts[1])
-	if err != nil {
-		return "", nil, fmt.Errorf("cannot decode base64 data: %v", err)
-	}
-	return string(parts[0]), buf[:n], nil
+	return encodeV1(buf.Bytes()), nil
 }
 
-func decodeOpenpgp(formatAndBase64 []byte, kind string) (packet.Packet, error) {
-	if len(formatAndBase64) == 0 {
+func decodeV1(b []byte, kind string) (packet.Packet, error) {
+	if len(b) == 0 {
 		return nil, fmt.Errorf("empty %s", kind)
 	}
-	format, data, err := splitFormatAndBase64Decode(formatAndBase64)
+	buf := make([]byte, base64.StdEncoding.DecodedLen(len(b)))
+	n, err := base64.StdEncoding.Decode(buf, b)
 	if err != nil {
-		return nil, fmt.Errorf("%s: %v", kind, err)
+		return nil, fmt.Errorf("%s: cannot decode base64 data: %v", kind, err)
+	}
+	if n == 0 {
+		return nil, fmt.Errorf("empty decoded %s", kind)
 	}
-	if format != "openpgp" {
-		return nil, fmt.Errorf("unsupported %s format: %q", kind, format)
+	buf = buf[:n]
+	if buf[0] != v1 {
+		return nil, fmt.Errorf("unsupported %s format version: %d", kind, buf[0])
 	}
-	pkt, err := packet.Read(bytes.NewReader(data))
+	rd := bytes.NewReader(buf[1:])
+	pkt, err := packet.Read(rd)
 	if err != nil {
 		return nil, fmt.Errorf("cannot decode %s data: %v", kind, err)
 	}
+	if rd.Len() != 0 {
+		return nil, fmt.Errorf("%s has spurious trailer data", kind)
+	}
 	return pkt, nil
 }
 
 // Signature is a cryptographic signature.
-type Signature interface {
-	// KeyID() returns a suffix of the signing key fingerprint
-	KeyID() string
-}
+type Signature interface{}
```

Comment:
That's not very useful. Anything would go there, unperceived.

Should we use a *Signature struct instead, with a private field?